agent/agent.go

```diff
@@ -345,6 +363,24 @@ func (a *Agent) connectionManager(ctx context.Context, cancel context.CancelFunc
 		a.connectedMu.Unlock()
 
 		go a.receiveLoop(ctx, cancel, conn)
+
+		// Block until we receive an entity config, or the grace period expires
+		select {
+		case <-a.entityConfigCh:
+			logger.Debug("successfully received the initial entity config")
+		case <-time.After(entityConfigGracePeriod):
+			logger.Warning("the initial entity config was never received, using the local entity")
+		case <-ctx.Done():
+			// The connection was closed before we received an entity config or we
+			// reached the grace period. We should wait before reconnecting in order
+			// to avoid flooding the backend
+			time.Sleep(time.Second * 1)
```

Comment:
Unfortunately no, because the backoff stuff is in `a.connectWithBackoff` and it immediately returns since it successfully connected; I'll try so see on the agent side if the validation could be done earlier.

Reply:
Nevermind, by moving the namespace validation stuff before the HTTP connection is upgraded to a WebSocket, I was able to make it behave like it should, so the retry w/ backoff mechanism will kick in